.github/workflows/ci.yml

```diff
@@ -232,6 +258,54 @@ jobs:
         password: ${{ secrets.PYPI_TOKEN }}
         verbose: true
 
+  coverage-report:
+    name: Coverage report
+    runs-on: ubuntu-latest
+    needs:
+      # We are waiting only for test jobs.
+      - test-linux
+      - test-windows
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Install dependencies
+      run: python -m pip install --upgrade pip coverage[toml] diff_cover
+
+    - name: Download coverage reports
+      uses: actions/download-artifact@v3
+      with:
+        name: coverage-${{ github.run_id }}
+        path: .
+
+    - name: Combine coverage
+      run: coverage combine
+
+    - name: Report coverage
+      run: |
+          coverage report --no-skip-covered --show-missing
+          coverage report --show-missing > coverage-report.txt
+
+    # Use the generic JS script to call our custom script
+    # for sending a comment to a PR.
+    - uses: actions/github-script@v3
+      env:
+        COMMENT_MARKER: "<!--- automatic-coverage-report -->"
+        COMMENT_BODY: coverage-report.txt
+      with:
+        script: |
+          const script = require(`${process.env.GITHUB_WORKSPACE}/.github/scripts/pr_comment.js`)
+          // Only pass top level objects as GHA does dependecy injection.
+          await script({github, context, process})
+
+    - name: Diff coverage
+      run: |
+        git fetch
```

Comment:
This is still needed after `fetch-depth: 0` above?

Reply:
I don't know. I think so. Let's see.

```suggestion
```